code/datums/components/human_damage_callback.dm

```diff
@@ -0,0 +1,26 @@
+/datum/component/human_damage_callback
+	dupe_type = COMPONENT_DUPE_ALLOWED
+	var/datum/callback/callback
+	var/list/allowed_damage_types
+	var/damage_limit
+	var/total_damage = 0
+
+// Takes a limit and a callback as manditory arguments. Will not unregister itself after the callback is called
+// Optionally takes the allowed damage types as argument in the form of (BRUTE, BURN) for example
+/datum/component/human_damage_callback/Initialize(damage_limit, callback, damage_types = null)
+	var/mob/living/carbon/human/H = parent
+	if(!istype(H) || damage_limit <= 0 || !callback) //Something went wrong
+		return COMPONENT_INCOMPATIBLE
+	src.damage_limit = damage_limit
+	src.callback = callback
```

Comment:
To quote a former maintainer. If you have to use `src.var_name = var_name`, then you should probably rename your vars.

Reply:
Will change once able